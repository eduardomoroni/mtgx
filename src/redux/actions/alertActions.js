// @flow

import { SHOW_MESSAGE, CLEAR_MESSAGE } from '../types'

export const showMessage = (message: string) => ({ type: SHOW_MESSAGE, message })
export const clearMessage = () => ({ type: CLEAR_MESSAGE })
