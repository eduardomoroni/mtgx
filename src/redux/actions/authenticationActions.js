// @flow

import { SET_USER } from '../types'

export const setUser = (user: Object) => ({ type: SET_USER, user })
