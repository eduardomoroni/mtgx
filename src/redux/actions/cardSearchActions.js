import { SHOW_MESSAGE, CLEAR_MESSAGE } from '../types'

export const showMessage = (message) => ({ type: SHOW_MESSAGE, message })
export const clearMessage = () => ({ type: CLEAR_MESSAGE })
