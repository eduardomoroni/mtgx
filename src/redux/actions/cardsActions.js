import { SET_CARDS } from '../types'

export const setCards = (cards) => ({ type: SET_CARDS, cards })
