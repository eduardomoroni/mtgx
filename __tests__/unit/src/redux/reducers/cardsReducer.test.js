import { setCards } from '../../../../../src/redux/actions/cardsActions'
import CardSearchReducer, { initialState } from '../../../../../src/redux/reducers/cardsReducer'

describe('CardSearch reducer', () => {
  it('should have a snapshot', async () => {
    expect(initialState).toMatchSnapshot()
  })

  it('SHOW_MESSAGE action type should set message on state', async () => {
    const cards = 'TODO: This should be a Map of cards'
    const action = setCards(cards)
    const state = CardSearchReducer(initialState, action)

    expect(state).toEqual(cards)
  })
})
