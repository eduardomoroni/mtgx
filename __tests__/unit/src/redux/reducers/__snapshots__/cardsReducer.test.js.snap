// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSearch reducer should have a snapshot 1`] = `""`;
